### ISSUE TITLE
maven plugin creates invalid testNG Citrus tests

### ISSUE THREAD

**Issue by Amr-Abouelleil-bose** [citrusframework/citrus#562]
**Citrus Version**
2.7.8

**Expected behavior**

Following the [maven citrus tutorial](https://citrusframework.org/docs/setup-maven/) should not cause an error when creating and running the final sample test in the tutorial. 

**Actual behavior**
```
21:50:25,724 INFO  port.LoggingReporter| C I T R U S  T E S T S  2.7.8
21:50:25,724 INFO  port.LoggingReporter| 
21:50:25,724 INFO  port.LoggingReporter| ------------------------------------------------------------------------
21:50:25,724 DEBUG port.LoggingReporter| BEFORE TEST SUITE
21:50:25,724 INFO  port.LoggingReporter| 
21:50:25,724 INFO  port.LoggingReporter| 
21:50:25,724 INFO  port.LoggingReporter| BEFORE TEST SUITE: SUCCESS
21:50:25,724 INFO  port.LoggingReporter| ------------------------------------------------------------------------
21:50:25,724 INFO  port.LoggingReporter| 
[Utils] [ERROR] [Error] org.testng.TestNGException: 
Cannot inject @Test annotated Method [myTest] with [interface com.consol.citrus.dsl.runner.TestRunner].
For more information on native dependency injection please refer to http://testng.org/doc/documentation-main.html#native-dependency-injection
	at org.testng.internal.Parameters.checkParameterTypes(Parameters.java:244)
	at org.testng.internal.Parameters.createParameters(Parameters.java:172)
	at org.testng.internal.Parameters.createParameters(Parameters.java:458)
	at org.testng.internal.Parameters.handleParameters(Parameters.java:569)
	at org.testng.internal.Invoker.handleParameters(Invoker.java:1308)
	at org.testng.internal.Invoker.createParameters(Invoker.java:1036)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1126)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)

21:50:30,757 INFO  port.LoggingReporter| 
```

**Test case sample**
> Please, share the test case (as small as possible) which shows the issue

```
package com.consol.citrus;

import com.consol.citrus.annotations.CitrusResource;
import com.consol.citrus.annotations.CitrusTest;
import com.consol.citrus.dsl.runner.TestRunner;
import com.consol.citrus.dsl.testng.TestNGCitrusTestRunner;
import org.testng.annotations.Test;

/**
 *
 *
 * @author Unknown
 * @since 2018-12-01
 */
public class MyTest extends TestNGCitrusTestRunner {
    @CitrusTest
    @Test
    public void myTest(@CitrusResource TestRunner testRunner) {
        testRunner.echo("TODO: Code the test MyTest");

    }
}
```

**Comment by svettwer** [comment]
Hi!

Thx for the report! :+1: 

The test should be setup like this:
```java
public class MyTest extends TestNGCitrusTest {

    @CitrusTest
    @Test @Parameters("context")
    public void myTest(@Optional @CitrusResource TestRunner testRunner) {
        testRunner.echo("TODO: Code the test MyTest");

    }
}
```

We'll fix the citrus maven plugin so that the generated tests are functional.
We should also review the [documentation concerning test context injection](http://citrusframework.org/citrus/reference/html/index.html#test-context-injection).

BR,
Sven

**Comment by Amr-Abouelleil-bose** [comment]
Hi Sven,

Thanks for your response. When I try to use the code snippet you provide, I find that echo is not resolvable method for testRunner. Any ideas?

**Comment by svettwer** [comment]
Hi!

This might be the case, if you're missing the `citrus-java-dsl` dependency.

```xml
    <dependency>
      <groupId>com.consol.citrus</groupId>
      <artifactId>citrus-java-dsl</artifactId>
      <version>${citrus.version}</version>
    </dependency>
```

I've checked the tutorial again. The archetype should have created that in case that you chose the java DSL. :thinking: 

BR,
Sven

**Comment by svettwer** [comment]
Closed with #599
